ql/src/java/org/apache/hadoop/hive/ql/queryhistory/repository/IcebergRepository.java

```diff
@@ -53,17 +55,41 @@
 import java.util.Map;
 import java.util.Queue;
 import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class IcebergRepository extends AbstractRepository implements QueryHistoryRepository {
   private static final Logger LOG = LoggerFactory.getLogger(IcebergRepository.class);
   private static final String ICEBERG_STORAGE_HANDLER = "org.apache.iceberg.mr.hive.HiveIcebergStorageHandler";
+  private static final String ICEBERG_WORKER_THREAD_NAME_FORMAT = "query-history-iceberg-worker-pool-%d";
 
   private HiveOutputFormat<?, ?> outputFormat;
   private Serializer serializer;
   @VisibleForTesting
   HiveStorageHandler storageHandler;
   @VisibleForTesting
   TableDesc tableDesc;
+  private ExecutorService icebergExecutor;
+
+  @Override
+  public void init(HiveConf conf, Schema schema) {
+    super.init(conf, schema);
+    icebergExecutor = Executors.newFixedThreadPool(getIcebergWorkerPoolSize(),
+        (new ThreadFactoryBuilder()).setDaemon(true).setNameFormat(ICEBERG_WORKER_THREAD_NAME_FORMAT).build());
+  }
+
+  /**
+   * Calculates the pool size for the ExecutorService to be passed to the committer.
+   * The thread pool adheres to the configuration settings for Iceberg worker pools.
+   * @return pool size for iceberg worker pool
+   */
+  private int getIcebergWorkerPoolSize() {
+    String value = System.getProperty("iceberg.worker.num-threads");
```

Comment:
you don't need for flush the same config as for iceberg worker pool. 2 threads should be enough

Reply:
okay, using 2 threads will keep this simpler